0000-event-system.md

```diff
@@ -0,0 +1,109 @@
+- Start Date: 2018-01-16
+- RFC PR: (leave this empty)
+- React Navigation Issue: (leave this empty)
+
+# Summary
+
+Add an event emitter to the navigation prop that emits focus/blur events for the route.
+
+This proposal has been pioneered by @satya164: https://github.com/react-navigation/react-navigation/issues/1363 https://github.com/react-navigation/react-navigation/issues/51
+
+# Basic example
+
+In this example, we focus the first text input when the screen has completed transitioning:
+
+```
+componentDidMount() {
+	this._sub = this.props.navigation.events.addListener(
+		'DidFocus',
```

Comment:
- I think event names should be camelcase with the first letter starting as lower case -- `didFocus` -- this is more of a standard for events in JS (eg: https://reactjs.org/docs/events.html#reference)
- Additionally, I think I personally prefer for the events to use the `on` prefix, such as `onFocus`, `onBlur`, `onWillFocus` (once again for consistency, see above link again)
- Lastly, I think we should hoist the `events.addListener` function up to the `navigation` object itself -- so we have `this.props.navigation.addEventListener`(`addEventListener` is used by AppState and NetInfo so the name is consistent). This is a smaller API surface area --
 only one property is exposed, `addEventListener`, rather than both `events` and `addListener` on events.

Reply:
Sure, these conventions seem more common. I'll fix up the RFC for that